qmf/src/plugins/messageservices/imap/imapstrategy.cpp

```diff
@@ -155,13 +155,14 @@ bool updateMessagesMetaData(ImapStrategyContextBase *context,
                                    const QMailMessageKey &storedKey, 
                                    const QMailMessageKey &unseenKey, 
                                    const QMailMessageKey &seenKey,
+                                   const QMailMessageKey &flaggedAsDeleted,
                                    const QMailMessageKey &flaggedKey,
                                    const QMailMessageKey &unreadElsewhereKey,
                                    const QMailMessageKey &importantElsewhereKey,
                                    const QMailMessageKey &unavailableKey)
 {
     bool result = true;
-    QMailMessageKey reportedKey(seenKey | unseenKey);
+    QMailMessageKey reportedKey((seenKey | unseenKey)  & ~flaggedAsDeleted);
```

Comment:
This has the effect of purging the messages in flaggedAsDeleted, right (they're also in storedKey because it includes deleted messages from the mailbox)?  I would have thought they should not be purged until they become expunged...